Feat: health check guide (#6592)
Kong/docs.konghq.com: app/_src/gateway/production/monitoring/healthcheck-probes.md

* Liveness: This `/status` endpoint responds with a `200 OK` status if Kong is running. The request will fail either with a `500 Internal Server Error` or no response if Kong is not running. You can send a GET request to check the liveness of your {{site.base_gateway}} instance:
  ```sh
  # Replace `localhost:8100` with the appropriate host and port for
  # your Status API server
  
  curl -i http://localhost:8100/status
  ```

* Readiness: This `/status/ready` endpoint responds with a `200 OK` status if Kong has successfully loaded a valid configuration and is ready to proxy traffic. The request will fail either with a `500 Internal Server Error` or no response if Kong is not ready to proxy traffic yet. You can send a GET request to check the readiness of your {{site.base_gateway}} instance:
  ```sh
  # Replace `localhost:8100` with the appropriate host and port for
  # your Status API server
  
  curl -i http://localhost:8100/status/ready
  ```

These two types of health checks for {{site.base_gateway}} are modeled on how [Kubernetes defines](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) health check probes.

> It is strongly recommended that a component (i.e. load balancer) perform the readiness health check before sending traffic. This ensures that a {{site.base_gateway}} node has not only successfully started up but has also finished loading up the configuration and is ready to receive proxy traffic. 

Reviewer: ```suggestion
We strongly recommend that a component (that is, a load balancer) perform the readiness health check before sending traffic. This ensures that a {{site.base_gateway}} node has not only successfully started up but has also finished loading up the configuration and is ready to receive proxy traffic. 
```
* "We recommend" is the phrasing we use in the docs (and I just found out that it wasn't in our style guide, so I've gone and added it).
* Avoid using latin shortforms.

> The liveness health check may return a 200 OK before the readiness health check does while it's running but still loading the full configuration. If a component only monitors the liveness probe to decide when to send traffic to {{site.base_gateway}}, then there will be a short period of time where requests will be met with a `404 Not Found` response the {{site.base_gateway}} is ready to proxy traffic. We recommend using the readiness probe over the liveness probe especially in production environments.

Reviewer: ```suggestion
The liveness health check may return a 200 OK response before the readiness health check does. 
Even if {{site.base_gateway}} is running, it may still be loading the full configuration, which means it is live but not ready.
If a component only monitors the liveness probe to decide when to send traffic to {{site.base_gateway}}, there will be a short period of time where requests will be met with a `404 Not Found` response before the {{site.base_gateway}} is ready to proxy traffic. 
We recommend using the readiness probe over the liveness probe, especially in production environments.
```
Some rephrasing for clarity.

status_listen = 0.0.0.0:8100
```

{:.note}

> > **Note:**  Readiness probes should be utilized on each node within the cluster, including standalone nodes; control plane nodes; and data plane nodes. Checking only one node in a cluser is insufficient.

Reviewer: ```suggestion
> **Note:** Readiness probes should be used on every node within the cluster, including standalone, control plane, and data plane nodes. Checking only one node in a cluster is insufficient.
```

    periodSeconds: 5
```

{:.note}

> > **Note:**  Failure to set an `initialDelaySeconds` may result in {{site.base_gateway}} entering a crash loop, as it requires a short time to fully load the configuration. The time to delay can depend on the size of the configuration.

Reviewer: ```suggestion
> **Note:** Failure to set an `initialDelaySeconds` may result in {{site.base_gateway}} entering a crash loop, as it requires a short time to fully load the configuration. The time to delay can depend on the size of the configuration.
```

> ## Using a readiness check in version 3.2 or lower.

Reviewer: ```suggestion
## Using a readiness check in version 3.2 or lower
```

> The `/status/ready` endpoint was added in version 3.3, meaning versions 3.2 and lower do not benefit from this built-in readiness endpoint. However a workaround exists which is strongly recommended:

Reviewer: ```suggestion
The `/status/ready` endpoint was added in version 3.3, so prior versions don't benefit from this built-in readiness endpoint. 
We recommend the following workaround for those versions:
```

> 1. Configure a new route in {{site.base_gateway}} with the path uniqely set for this purpose. For example, a path of `/health/ready` can be used. No need for a service, only the route is required.

Reviewer: ```suggestion
1. Configure a new route in {{site.base_gateway}} with the path uniquely set for this purpose. This route doesn't require a service.

   For example, you could use the path `/health/ready`.
```

2. Configure the Request Termination plugin to respond to requests on that route with a HTTP 200 status code.

> Note that in this workaround, the port to send health-check requests to will be the proxy port (8000 & 8443 by default) instead of the status API port.

Reviewer: ```suggestion
{:.note}
> **Note:** In this workaround, the port to send health check requests to is the proxy port (8000 & 8443 by default) instead of the status API port.
```

> ## What is not covered by these health checks

Reviewer: ```suggestion
## What isn't covered by health checks?
```

> It is important to understand that a health check probe does not take into account the following:

Reviewer: ```suggestion
A health check probe doesn't take the following into account:
```

* If {{site.base_gateway}} is performing optimally or not

> * If {{site.base_gateway}} is throwing intermittent failures due to any reason

Reviewer: ```suggestion
* If {{site.base_gateway}} is throwing intermittent failures for any reason
```

> * If {{site.base_gateway}} is throwing errors due to third-party systems like DNS, cloud provider outages, network failures, etc.

Reviewer: ```suggestion
* If {{site.base_gateway}} is throwing errors due to third-party systems like DNS, cloud provider outages, network failures, and so on
```

> * If any upstream services are throwing errors or responding too slow

Reviewer: ```suggestion
* If any upstream services are throwing errors or responding too slowly
```

## See also

For more information on Kong and related topics, check out the following resources:

> * [Health Check / Monitoring Overview](/gateway/latest/production/monitoring/)

Reviewer: ```suggestion
* [Health Check and Monitoring Overview](/gateway/latest/production/monitoring/)
```
